Add notes I took during the 1.2-alpha.2 release party (#424)
cert-manager/website: content/en/docs/contributing/release-process.md

First ensure that you have all the tools and permissions required to perform a cert-manager release:

1. Install the [`release-notes`]((https://github.com/kubernetes/release/blob/master/cmd/release-notes/README.md)) CLI:
   ```sh
   (cd && GO111MODULE=on go get k8s.io/release/cmd/release-notes@latest)

>    ```

Reviewer: This is what I do, but below you talk about the `krel release-notes` subcommand, which may be the proper way to use these tools.
Make it consistent anyway.

2. Install our [`cmrel`](https://github.com/cert-manager/release) CLI:
   ```sh
   (cd && GO111MODULE=on go get github.com/cert-manager/release/cmd/cmrel@latest)
   ```

> 3. Get the "Cloud Build Editor" permission for the

Reviewer 2: Not so important, but perhaps it's also worth to get view permissions for Storage- in case there is a need to manually check the releases bucket.

release are outlined as part of a GitHub milestone. We cut a release even if
some of these goals are missed, in order to keep up release velocity.

### Process for releasing a minor version

> Note: please click on the "Edit this page" button if a step is missing or

> > if it oudated.

Reviewer: ```suggestion
> Note: please click on the "Edit this page" button if a step is missing or
> if it is outdated.
```

The process for cutting a minor release is as follows:

1. Ensure upgrading document exists. See for example, see
   [upgrading-1.0-1.1](https://cert-manager.io/docs/installation/upgrading/upgrading-1.0-1.1/)
   (not necessary for alpha and patch releases)

2. Create or update the release branch

>     If this is the first (`alpha1`) release, then you will need to create
>     the release branch:

Reviewer: ```suggestion
    If this is the first alpha release (`alpha.0`), then you will need to create
    the release branch:
```

    ```bash
    git fetch --all
    git checkout -b release-1.0 origin/master
    ```

>     If there has already been an `alpha1` release, the release branch will
>     already exist, so you will need to update it with the latest commits
>     from the master branch, as follows:

Reviewer: ```suggestion
    If there has already been an alpha release, the release branch will
    already exist, so you will need to update it with the latest commits
    from the master branch, as follows:
```

    git push --set-upstream origin
    ```

>     > ⚠️ You need to be an "admin" to be able to push to `release-1.0`.

Reviewer: ```suggestion
    > ⚠️ You need to be a GitHub project "admin" to be able to push to the release branch
```

4. [Generate and edit the release notes](#generating-and-editing-the-release-notes)

5. Run `cmrel stage`

   1. In this example we stage a release using the 'release-1.0' branch,
      setting  the release version to `v1.0.0`:

      ```bash
      cmrel stage --branch=release-1.0 --release-version=v1.0.0
      ```

      This step takes ~10 minutes. It will build all Docker images and
      create all the manifest files and upload them to a storage bucket on
      Google Cloud. These artifacts will be published and released in the
      next steps.

      The final line of output contains URL of the bucket
      containing the release artifacts. The final segment in that URL
      contains the `RELEASE_NAME`, which you will need in the next step.

   2. While the build is running, send a first Slack message to
      `#cert-manager-dev`:

      > Releasing 1.2.0-alpha.2 🧵

>    3. Send a second Slack message in reply to this first message with the

Reviewer 2: Would it be worth to add here a warning to check at this point whether the Cloud Build output doesn't contain any sensitive data?

        You can view the progress by clicking the Google Cloud Build URL in the output of this command.

   2. While the build is running, send a third Slack message in reply to
      the first message:

      > Publish dry-run step: <https://console.cloud.google.com/cloud-build/builds/16f6f875-0a23-4fff-b24d-3de0af207463?project=1021342095237>

   3. Next publish the release artifacts for real.

      If the last step succeeded, you can now re-run the `cmrel publish`
      with the `--nomock` argument to actually publish the release
      artifacts to GitHub, `Quay.io`, [Helm Hub] etc.

      ```bash
      cmrel publish --nomock --release-name <RELEASE_NAME>
      ```

      NOTE: At this stage there will be a draft release on GitHub and a
      live release on [Helm Hub]. So you must now complete the release

>       process quickly otherwise users of the latest release on[Helm Hub]

Reviewer 2: [Helm Hub] - I think there is a broken/missing link, probably should be pointing to `https://artifacthub.io/packages/helm/jetstack/cert-manager` ? Also the name seems to be ArtifactHub now.

Author: Yeah, https://charts.jetstack.io returns a 404 instead of 301 to https://artifacthub.io/packages/helm/jetstack/cert-manager... I investigated in https://github.com/jetstack/cert-manager/issues/3439#issuecomment-768898542 

Can I rename "Helm Hub" to "Helm chart"?

      will encounter errors, because the manual CRD install URL will not be
      available yet.

   4. While the build is running, send a fourth Slack message in reply to
      the first message:

      > Publish step: <https://console.cloud.google.com/cloud-build/builds/b6fef12b-2e81-4486-9f1f-d00592351789?project=1021342095237>

7. Publish the GitHub release

   1. Visit the draft GitHub release and paste in the release notes that you generated earlier.

      You will need to manually edit the content to match the style of earlier releases.
      In particular:

      * Remove package related changes
      * Replace links to `@jetstack-bot`, in cherry-pick PRs, with links to

>         the GitHub handle of the author of the original PR.

Reviewer: This didn't seem to happen with the latest version of `krel release-notes` -- the release notes had the correct authors -- so perhaps we can drop this line.

Patch releases are cut on an ad-hoc basis, depending on recent activity on the
release branch.

### Process for releasing a patch version

The process for cutting a patch release is as follows:

1. Ensure that all PRs have been cherry-picked into the release branch, e.g. `release-1.0`

   Bugs that need to be fixed in a patch release should be [cherry picked into the appropriate release branch](../contributing-flow/#cherry-picking).

> 2. Then, continue with the instruction in [process for releasing a minor version](#process-for-releasing-a-minor-version).

Reviewer: ```suggestion
2. Then, continue with the instructions in [process for releasing a minor version](#process-for-releasing-a-minor-version).
```

## Generating and editing the release notes

1. Run as follows, substituting the current and last versions where appropriate:

    ```bash
    export GITHUB_TOKEN=*your-token*
    export RELEASE_VERSION=1.0.0
    export BRANCH=release-1.0
    export END_REV=release-1.0
    export START_REV=v0.16.1

>     krel release-notes --github-repo cert-manager --github-org jetstack --required-author "jetstack-bot" --output release-notes.md

Reviewer: This is what I referred to above. If this is the correct way to use the command, then I think the installation instructions may need adjusting to help users install `krel`.

Author: oops, `krel` is the wrong tool:( It should be release-notes

    ```

>     NOTE: The GitHub token needs read-only permission to the cert-manager repository.

Reviewer 2: Nit: In the Prerequisites section it says that the token 'does not need any privileges'. I guess since the cert-manager repository is public, no-privileges and read-only permissions are equivalent? But maybe it's worth to use the same wording in both places.
